pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -66,36 +69,78 @@ func (am *Alertmanager) Run(ctx context.Context) error {
 		return errors.Wrap(err, "unable to initialize the silencing component of alerting")
 	}
 
-	am.notificationLog, err = nflog.New(
-		nflog.WithRetention(time.Hour*24),                         //TODO: This is a setting.
-		nflog.WithSnapshot(filepath.Join("dir", "notifications")), //TODO: This should be a setting
-	)
-	if err != nil {
-		return errors.Wrap(err, "unable to initialize the notification log component of alerting")
+	return nil
+}
+
+func (am *Alertmanager) Run(ctx context.Context) error {
+	// Make sure dispatcher starts. We can tolerate future reload failures.
+	if err := am.ReloadConfigFromDatabase(); err != nil {
```

Comment:
I've already implemented this as part of #32107, I'm happy to merge as is then I'll rebase and fix.

Reply:
I will merge this and rebase that PR